examples/vault-cluster-enterprise/startup-script-vault.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# This script is meant to be run as the Startup Script of each Compute Instance while it's booting. The script uses the
+# run-consul and run-vault scripts to configure and start both Vault and Consul in client mode. This script assumes it's
+# running in a Compute Instance based on a Google Image built from the Packer template in
+# examples/vault-consul-image/vault-consul.json.
+
+set -e
+
+# Send the log output from this script to startup-script.log, syslog, and the console
+# Inspired by https://alestic.com/2010/12/ec2-user-data-output/
+exec > >(tee /var/log/startup-script.log|logger -t startup-script -s 2>/dev/console) 2>&1
+
+# The Packer template puts the TLS certs in these file paths
+readonly VAULT_TLS_CERT_FILE="/opt/vault/tls/vault.crt.pem"
+readonly VAULT_TLS_KEY_FILE="/opt/vault/tls/vault.key.pem"
+
+# Note that any variables below with <dollar-sign><curly-brace><var-name><curly-brace> are expected to be interpolated by Terraform.
+/opt/consul/bin/run-consul --client --cluster-tag-name "${consul_cluster_tag_name}"
+/opt/vault/bin/run-vault --gcs-bucket ${vault_cluster_tag_name} \
+  --tls-cert-file "$VAULT_TLS_CERT_FILE" \
+  --tls-key-file "$VAULT_TLS_KEY_FILE" \
+  --enable-auto-unseal \
+  --auto-unseal-project "${vault_auto_unseal_project_id}" \
+  --auto-unseal-region "${vault_auto_unseal_region}" \
+  --auto-unseal-key-ring "${vault_auto_unseal_key_ring}" \
+  --auto-unseal-crypto-key "${vault_auto_unseal_crypto_key}"
```

Comment:
This interface looks really nice. It's very clear what's happening here.

Reply:
ok cool